### PR TITLE
Best practices > Troubleshooting latency

### PR DESCRIPTION
**Pull request by jessiemongeon1**
Thank you for your contribution to the IC Developer Portal.
Before submitting your Pull Request, please make sure that:

- [ ] You have updated the [`.github/CODEOWNERS`](https://github.com/dfinity/portal/blob/master/.github/CODEOWNERS) file according to your changes.
- [ ] You have registered new documents to [`/sidebars.js`](https://github.com/dfinity/portal/blob/master/sidebars.js) if any.

### CODE REVIEW

File: docs/developer-docs/setup/best-practices/troubleshooting.mdx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+When a dapp or canister is running with a high amount of latency, there are some steps that developers can take to troubleshoot or optimize the performance of their canisters. 
+
+This document will detail some troubleshooting methods and performance optimizations that developers can use when troubleshooting latency. 
+
+## Troubleshooting best practices 
+
+### Recommendation: Use query calls instead of update calls when possible.
+
+[Query calls](/docs/current/developer-docs/backend/design-dapps#using-query-calls) are executed on a single node within a subnet and do not go through consensus, therefore they can be returned much faster than an update call. For some applications, query calls can be used in place of update calls to reduce latency.
```
Comment (jennifertrin) [inline docs/developer-docs/setup/best-practices/troubleshooting.mdx:13]: We can consider adding this note from Andriy's blog post: https://internetcomputer.org/blog/features/improving-query-latencies#takeaways-how-to-improve-query-latencies

**Review by dfx-json — APPROVED** [review]

**Review by dsarlis — APPROVED** [review]

File: docs/developer-docs/setup/best-practices/troubleshooting.mdx (hunk 1 above)
Comment (jessiemongeon1) [reply docs/developer-docs/setup/best-practices/troubleshooting.mdx:13]: Added a section on efficient query calls

**Comment by jessiemongeon1** [comment]
> 2. Should we mention calling an [actor class]((https://internetcomputer.org/docs/current/motoko/main/actor-classes) constructor is similar to installing a fresh canister on the platform?

Since this is Motoko-specific, I think this would fit better in the Best Practices > General page in the Motoko section.

**Comment by jennifertrin** [comment]
> > 2. Should we mention calling an [actor class]((https://internetcomputer.org/docs/current/motoko/main/actor-classes) constructor is similar to installing a fresh canister on the platform?
> 
> Since this is Motoko-specific, I think this would fit better in the Best Practices > General page in the Motoko section.

Good point. Maybe we should make a note of this?

**Review by jennifertrin — APPROVED** [review]
Great work!

**Pull request merged**